### PR TITLE
Title: Change record api call frequency based on gps update

### PR DESCRIPTION
**Pull request by balsikandar-shuttl**

### CODE REVIEW

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt [hunk 1]
```diff
@@ -138,16 +143,40 @@ class LocationPingService : Service() {
...
+                if (timerTask == null) {
+                    initializeTimerTask()
+                }
+                if (longTimer == null) {
+                    longTimer = Timer()
+                }
+                longTimer?.schedule(
+                    timerTask,
+                    0,
+                    configs.minSyncInterval.toLong() / minSyncMultiplier
```
Comment (balsikandar-shuttl) [inline locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: replace / with * and make  multiplier float

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt [hunk 2]
```diff
@@ -157,6 +186,30 @@ class LocationPingService : Service() {
...
+            override fun run() {
+                pingLocations()
+            }
+        }
+    }
+
+    private fun initializeTimer() {
+        timer = object : CountDownTimer(
+            configs.timeout.toLong(),
+            configs.minSyncInterval.toLong() / minSyncMultiplier
```
Comment (balsikandar-shuttl) [inline locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: replace / with *

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt [hunk 3]
```diff
@@ -206,4 +259,21 @@ class LocationPingService : Service() {
...
+        }
+        mHandler.sendMessage(message)
+    }
+
+    private val mHandler = Handler(Looper.getMainLooper()) {
+        if (it.arg1 == 1001) {
+            stopWork()
+            Handler().postDelayed({
+                work()
+            }, 5 * 60 * 1000)
```
Comment (balsikandar-shuttl) [inline locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: use inactivity interval in timertask

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt [hunk 4]
```diff
@@ -69,7 +52,29 @@ class LocationPingService : Service() {
 
     private fun pingLocations() {
         try {
-            LocationRepo(LocationsDB.create(applicationContext)?.locationsDao()).syncLocations(
+            val locationsDao = LocationsDB.create(applicationContext)?.locationsDao()
+            val locationRepo = LocationRepo(locationsDao)
+
+            val locations = locationsDao?.getLimitedLocations(configs.batchSize)
```
Comment (balsikandar-shuttl) [inline locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: fetch last synced location only

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt (hunk 1 above)
Comment (balsikandar-shuttl) [reply locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: changed logic itself so not needed anymore

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt (hunk 2 above)
Comment (balsikandar-shuttl) [reply locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: not needed anymore

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt (hunk 3 above)
Comment (balsikandar-shuttl) [reply locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt]: done

File: locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt [hunk 5]
```diff
@@ -55,22 +55,22 @@ class LocationPingService : Service() {
...
                 val currentTime = System.currentTimeMillis() / 1000
 
-                Log.e("Debug", "dateTime $lastSaveTime, currentTime $currentTime")
+                Log.e("Debug", "dateTime $lastSaveTime, currentTime $currentTime, diff ${currentTime - lastSaveTime}")
 
-                if ((currentTime - lastSaveTime) > configs.inactivitySyncInterval) {
-                    minSyncMultiplier = 3
+                if ((currentTime - lastSaveTime) > configs.inactivityInterval/1000) {
+                    isLocationStale = true
                     sendMessage()
```
Comment (balsikandar-shuttl) [inline locationsyncservice/src/main/java/com/shuttl/location_pings/service/LocationPingService.kt:68]: rename to restartimers

**Review by betaCode95 — APPROVED** [review]

**Pull request merged**